src/playlist.rs

```diff
@@ -977,8 +844,64 @@ mod test {
                 fourth_song.to_owned(),
                 second_song.to_owned()
             ],
-            closest_album_to_group_by_key(group, pool.to_owned(), |s| s.bliss_song.to_owned())
-                .unwrap(),
+            closest_album_to_group(group, pool.to_owned()).unwrap(),
         );
     }
+
+    #[test]
+    fn test_forest_options() {
+        let first_song = Song {
+            path: Path::new("path-to-first").to_path_buf(),
+            analysis: Analysis::new([
+                1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
+            ]),
+            ..Default::default()
+        };
+        let first_song_dupe = Song {
+            path: Path::new("path-to-dupe").to_path_buf(),
+            analysis: Analysis::new([
+                1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
+            ]),
+            ..Default::default()
+        };
+
+        let second_song = Song {
+            path: Path::new("path-to-second").to_path_buf(),
+            analysis: Analysis::new([
+                2., 2., 2., 2., 2., 2., 2., 2., 2., 2., 2., 2., 2., 2., 2., 2., 1.9, 1., 1., 1.,
+            ]),
+            ..Default::default()
+        };
+        let third_song = Song {
+            path: Path::new("path-to-third").to_path_buf(),
+            analysis: Analysis::new([
+                2., 2., 2., 2., 2., 2., 2., 2., 2., 2., 2., 2., 2., 2., 2., 2., 2.5, 1., 1., 1.,
+            ]),
+            ..Default::default()
+        };
+        let fourth_song = Song {
+            path: Path::new("path-to-fourth").to_path_buf(),
+            analysis: Analysis::new([
+                2., 2., 2., 2., 2., 2., 2., 2., 2., 2., 2., 2., 2., 2., 2., 2., 0., 1., 1., 1.,
+            ]),
+            ..Default::default()
+        };
+        let fifth_song = Song {
+            path: Path::new("path-to-fifth").to_path_buf(),
+            analysis: Analysis::new([
+                2., 2., 2., 2., 2., 2., 2., 2., 2., 2., 2., 2., 2., 2., 2., 2., 0., 1., 1., 1.,
+            ]),
+            ..Default::default()
+        };
+
+        let mut songs = [
+            &first_song,
+            &first_song_dupe,
+            &second_song,
+            &third_song,
+            &fourth_song,
+            &fifth_song,
+        ];
+        closest_to_songs(&[&first_song], &mut songs, &ForestOptions::default());
```

Comment:
I believe we need an assert of some sorts here, to make sure we get the right playlist, no?

Reply:
Doh! I pushed a fix.

Reply:
thanks 😁 a little nitpick, but would it be possible to change it by scrambling a bit there order of `songs`?
Right now songs is not modified because it is already in the right order, would be nice to see it actually doing something :)
I'm thinking having
```
let mut songs = [
            &fourth_song,
            &second_song,
            &first_song,
            &third_song,
            &first_song_dupe,
            &fifth_song,
        ];
```
as the `songs` declaration for instance would do the job :)

Reply:
Okay, so it turns out that this test case wasn't working anyway. There were two issues:
* The algorithm doesn't really work with a single song as training data
* There is an element of randomness in the algorithm, so it's not possible to compare against a pre-defined expected order

I've added a test case with analysis of real tracks from my library instead. The tracks come from two Mozart piano concertos, and one Miles Davis record. The assertion is that the 5 songs of the Miles Davis record ends up last, but that the order of those songs don't matter.

I've run 10k rounds of passing tests now, so although this test case is non-deterministic and could fail in principle, it doesn't seem to happen often.

As for the issue that this metric doesn't really work with a single song as source, I guess that's a bit of a foot gun... I think the best thing to do is to add some warning about this in blissify?

Reply:
Makes sense! I added a few more comments, but they are mostly nits :smile: 

> As for the issue that this metric doesn't really work with a single song as source, I guess that's a bit of a foot gun... I think the best thing to do is to add some warning about this in blissify?

I think it would make sense to add this both in the documentation of the metric itself in bliss, as well as in blissify, yep! :)
Solving this issue.

Reply:
I've added a comment on this under the `DistanceMetricBuilder` trait. It seems like it's not possible to document this on the implementation on `ForestOptions` specifically, unfortunately.